Test.cs

```diff
@@ -12,7 +14,12 @@ public static long calculate(long number)
                 throw new Exception("Number must be non-negative.");
             }
 
-            return number == 0 ? 1 : number * Calculate(number + 1);
+            return number == 0 ? 1 : number * Calculate(number - 1);
+        }
```

Comment:
The method `Calculate` should return `number * Calculate(number - 1)` instead of `number * Calculate(number + 1)` to correctly implement the factorial calculation. This ensures the recursion progresses towards the base case of `number == 0`.